modules/samples/firealarm/component/manager/src/main/java/org/homeautomation/firealarm/manager/api/ManagerService.java

```diff
@@ -0,0 +1,337 @@
+            enrolmentInfo.setOwner(owner);
+            device.setEnrolmentInfo(enrolmentInfo);
+            KeyGenerationUtil.createApplicationKeys(DeviceTypeConstants.DEVICE_TYPE);
+            TokenClient accessTokenClient = new TokenClient(DeviceTypeConstants.DEVICE_TYPE);
+            AccessTokenInfo accessTokenInfo = accessTokenClient.getAccessToken(owner, deviceId);
+
+            //create token
+            String accessToken = accessTokenInfo.getAccess_token();
+            String refreshToken = accessTokenInfo.getRefresh_token();
+            List<Device.Property> properties = new ArrayList<>();
+            Device.Property accessTokenProperty = new Device.Property();
+            accessTokenProperty.setName("accessToken");
+            accessTokenProperty.setValue(accessToken);
+            Device.Property refreshTokenProperty = new Device.Property();
+            refreshTokenProperty.setName("refreshToken");
+            refreshTokenProperty.setValue(refreshToken);
+            properties.add(accessTokenProperty);
+            properties.add(refreshTokenProperty);
+            device.setProperties(properties);
+
+            boolean added = APIUtil.getDeviceManagementService().enrollDevice(device);
+            if (added) {
+                response.setStatus(Response.Status.OK.getStatusCode());
+            } else {
+                response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+            return added;
+        } catch (DeviceManagementException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error(e.getErrorMessage(), e);
+            return false;
+        } catch (AccessTokenException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error("Unable to obtain access token", e);
+            return false;
+        }
+    }
+
+    /**
+     * Remove installed device
+     *
+     * @param deviceId unique identifier for device
+     * @param response to request
+     */
+    @Path("manager/device/remove/{device_id}")
+    @DELETE
+    public void removeDevice(@PathParam("device_id") String deviceId,
+                             @Context HttpServletResponse response) {
+        DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+        deviceIdentifier.setId(deviceId);
+        deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+        try {
+            boolean removed = APIUtil.getDeviceManagementService().disenrollDevice(
+                    deviceIdentifier);
+            if (removed) {
+                response.setStatus(Response.Status.OK.getStatusCode());
+            } else {
+                response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+        } catch (DeviceManagementException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```

Comment:
The standard now is to return a Response object with a status code set and a payload attached if any.
Change the code accordingly. 

And also, it's better to log the error at server-side in case of an exception, so that it would be a good piece of information for system administrators to debug issues.